onnxruntime/core/providers/openvino/openvino_provider_factory.cc

```diff
@@ -184,13 +194,17 @@ struct OpenVINO_Provider : Provider {
   void* GetInfo() override { return &info_; }
 
   std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory(const void* void_params) override {
-    // Extract the void_params into ProviderOptions and ConfigOptions
-    using ConfigBuffer = std::pair<const ProviderOptions*, const ConfigOptions&>;
-    const ConfigBuffer* buffer = reinterpret_cast<const ConfigBuffer*>(void_params);
-    const auto& provider_options = *buffer->first;
-    const auto& config_options = buffer->second;
+    if (void_params == nullptr) {
+      LOGS_DEFAULT(ERROR) << "[OpenVINO EP] Passed NULL options to CreateExecutionProviderFactory()";
+      return nullptr;
+    }
+
+    std::array<void*, 2> pointers_array = *reinterpret_cast<const std::array<void*, 2>*>(void_params);
+    const ProviderOptions provider_options = *reinterpret_cast<ProviderOptions*>(pointers_array[0]);
```

Comment:
in latest commit, this line changed to make a copy of the ProviderOptions object. Is this done intentionally?